### PR TITLE
New PR for security group Rules

### PR DESCRIPTION
**Pull request by umarali-nagoor**
Test results

Apply complete! Resources: 6 added, 0 changed, 0 destroyed.

Outputs:

sg_rules = [
  {
    "code" = 0
    "direction" = "inbound"
    "ip_version" = "ipv4"
    "port_max" = 0
    "port_min" = 0
    "protocol" = "all"
    "remote" = "127.0.0.1"
    "rule_id" = "r006-ddfd256c-b3e7-4b97-9794-f16590bef872"
    "type" = 0
  },
  {
    "code" = 0
    "direction" = "inbound"
    "ip_version" = "ipv4"
    "port_max" = 807
    "port_min" = 805
    "protocol" = "udp"
    "remote" = "127.0.0.1"
    "rule_id" = "r006-053ba215-813b-4242-8201-f3bf38a7fd0a"
    "type" = 0
  },
  {
    "code" = 0
    "direction" = "inbound"
    "ip_version" = "ipv4"
    "port_max" = 22
    "port_min" = 22
    "protocol" = "tcp"
    "remote" = "0.0.0.0/0"
    "rule_id" = "r006-e540c4a8-37a5-490b-a2b3-4a21c042adb3"
    "type" = 0
  },
  {
    "code" = 0
    "direction" = "inbound"
    "ip_version" = "ipv4"
    "port_max" = 0
    "port_min" = 0
    "protocol" = "icmp"
    "remote" = "127.0.0.9/0"
    "rule_id" = "r006-378c5d6c-f44c-46b9-98fe-49ce4d13ac15"
    "type" = 8
  },
]

### CODE REVIEW

File: website/docs/d/is_security_group_rule.html.markdown [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+data "ibm_is_security_group_rule" "sg1_rule" {
+  name = ibm_is_security_group.testacc_security_group.name
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `group` - (Required, Forces new resource, string) The security group id.
```
Comment (hkantare) [inline website/docs/d/is_security_group_rule.html.markdown:73]: why all these attributes are under argment its the datasource not the resource...only name is required parameter

remaining all are computed

File: website/docs/d/is_security_group_rule.html.markdown [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+  Manages IBM Security Group Rule.
+---
+
+# ibm\_is_security_group_rule
+
+Provides a security group rule data source. This allows security group rule to be read.
+
+
+## Example Usage
+
```
Comment (hkantare) [inline website/docs/d/is_security_group_rule.html.markdown:15]: This is a datasource not resource change the 
In the following example, you can create a different types of protocol rules `ALL`, `ICMP`, `UDP` and `TCP`
In the following example, you can create a different types of protocol rules `ALL`, `ICMP`, `UDP` and `TCP` and read the security group

File: ibm/provider.go [hunk 3]
```diff
@@ -202,6 +202,7 @@ func Provider() terraform.ResourceProvider {
 			"ibm_is_ssh_key":                       dataSourceIBMISSSHKey(),
 			"ibm_is_subnet":                        dataSourceIBMISSubnet(),
 			"ibm_is_subnets":                       dataSourceIBMISSubnets(),
+			"ibm_is_security_group_rule":           dataSourceIBMISSecurityGroupRule(),
```
Comment (hkantare) [inline ibm/provider.go:205]: This datasource is for security group not group_rule

"ibm_is_security_group":           dataSourceIBMISSecurityGroup(),

Change the files name and method names to just refer as ibm_is_security_group

**Comment by umarali-nagoor** [comment]
Results:

Screenshot 2020-07-21 at 6 03 19 PM
Screenshot 2020-07-21 at 7 36 38 PM

**Pull request merged**